src/project_types/project.rs

```diff
@@ -154,20 +168,31 @@ impl Project {
         add: bool
     ) -> Result<(), Error> {
     
-        for mut u in &mut self.users {
-            if u.name == name {
-                if add {
-                    u.add_permission(perm);
-                } else {
-                    u.remove_permission(perm);
+        for i in 0..self.users.len() {
```

Comment:
'for i in self.users`. Might need a `.iter()` on there, I'm not sure if Rust implicits that. You should double check for any other `for i in 0..v.len()` instances. If you need the index number inside the loop, use `.iter().enumerate()`.